src/portable/portable.c

```diff
@@ -994,15 +1058,30 @@ static int install_chroot_dropin(
                                "LogExtraFields=PORTABLE=", base_name, "\n"))
                         return -ENOMEM;
 
+                /* If we have a single image then PORTABLE= will point to it, so we add PORTABLE_ID= with
+                 * the os-release fields and we are done. But if we have extensions, PORTABLE= will point
+                 * to the image where the current unit was found in. So we also list PORTABLE_ROOT= and
+                 * PORTABLE_ROOT_ID= for the base image, and PORTABLE_EXTENSION= and PORTABLE_EXTENSION_ID=
+                 * for each extension, so that all needed metadata is available. */
                 if (!ordered_hashmap_isempty(extension_images)) {
```

Comment:
invert the if check, i.e. put the empty case first, and the non-empty case second.